mrmustard/physics/ansatze.py

```diff
@@ -139,9 +145,9 @@ class PolyExpBase(Ansatz):
     """
 
     def __init__(self, mat: Batch[Matrix], vec: Batch[Vector], array: Batch[Tensor]):
-        self.mat = math.atleast_3d(math.astensor(mat))
-        self.vec = math.atleast_2d(math.astensor(vec))
-        self.array = math.atleast_1d(math.astensor(array))
+        self.mat = math.atleast_3d(mat)
+        self.vec = math.atleast_2d(vec)
+        self.array = math.atleast_1d(array)
```

Comment:
No, that is what I thought as well. But after discussions with Filippo we agreed that this is actually the best and also simplest way. No matter what we will always have to give an explicit batch dimension for c, since the polynomials can have any degree.